### PR TITLE
Fix #2262 Cover the frontend reducers with flow types.

### LINKED ISSUE #2262: Start covering the frontend with flow types.
I'm going to shoot for 60% on this first issue -- I have no idea how easy or hard this will be.

### PR DESCRIPTION
**Pull request by fzzzy**
This gives us 59% coverage; my goal was 60%. Close enough for now. I'll
cover some more of the frontend later.

### CODE REVIEW

File: frontend/src/app/reducers/addon.js [hunk 1]
```diff
@@ -1,22 +1,67 @@
...
+
+import type { Experiment } from './experiments';
+
+type RestartState = {
+  isRequired: boolean,
+};
+
+type AddonState = {
+  hasAddon: boolean,
+  installed: any,
```
Comment (fzzzy) [inline frontend/src/app/reducers/addon.js]: Maybe shouldn't use any type here

**Comment by fzzzy** [comment]
Covered the frontend actions, coverage up to 61%! Hit my initial goal.

File: frontend/test/app/reducers/newsletter-form-test.js [hunk 2]
```diff
@@ -46,7 +47,8 @@ describe('app/reducers/newsletter-form', () => {
   });
 
   describe('setFailed', () => {
-    const reduced = setFailed({});
+    const reduced = setFailed(defaultState());
+    console.log("setFailed", reduced);
```
Comment (fzzzy) [inline frontend/test/app/reducers/newsletter-form-test.js]: Whoops, need to remove this

File: flow-typed/seedrandom.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/*
+ * This Source Code is subject to the terms of the Mozilla Public License
+ * version 2.0 (the 'License'). You can obtain a copy of the License at
+ * http://mozilla.org/MPL/2.0/.
+ */
+/* eslint-disable */
```
Comment (pdehaan) [inline flow-typed/seedrandom.js]: Do you just want to add the "flow-typed/" directory to the .eslintignore file?

**Review by dannycoates — APPROVED** [review]
Sorry I took so long. This looks awesome!

**Pull request merged**